temporal_sqlalchemy/bases.py

```diff
@@ -9,18 +9,71 @@
 import sqlalchemy as sa
 import sqlalchemy.dialects.postgresql as sap
 import sqlalchemy.orm as orm
+import sqlalchemy.orm.base as base
 import sqlalchemy.orm.attributes as attributes
 import psycopg2.extras as psql_extras
 
 from temporal_sqlalchemy import nine
-from temporal_sqlalchemy.metadata import get_session_metadata
 
-_ClockSet = collections.namedtuple('_ClockSet', ('effective', 'vclock'))
+_PersistentClockPair = collections.namedtuple('_PersistentClockPairs',
+                                              ('effective', 'vclock'))
 
 T_PROPS = typing.TypeVar(
     'T_PROP', orm.RelationshipProperty, orm.ColumnProperty)
 
 
+class ActivityState:
+    def __set__(self, instance, value):
+        assert instance.temporal_options.activity_cls, "Make this better Joey"
```

Comment:
Can we change this error message to something a bit more helpful? 🙂  Perhaps:

```py
if instance.temporal_options.activity_cls:
    raise RuntimeError(
        "Can't set activity state on instance of %r because the activity class was never set."
        % type(instance).__name__)
```